pkg/ddevapp/ddevapp.go

```diff
@@ -563,33 +561,26 @@ func (app *DdevApp) GetWebserverType() string {
 	return v
 }
 
-// GetRouterHTTPPort returns app's router http port
-func (app *DdevApp) GetRouterHTTPPort() string {
-
-	// If the web container is running and HTTP_EXPOSE has a mapping,
-	// return the host-side mapped port
-	if httpExpose := app.GetWebEnvVar("HTTP_EXPOSE"); httpExpose != "" {
-		//util.Debug("GetRouterHTTPPort(): HTTP_EXPOSE=%s", httpExpose)
-		httpPort := app.PortFromExposeVariable(httpExpose, "80")
-		if httpPort != "" {
-			//util.Debug("GetRouterHTTPPort(): returning httpPort=%s found from HTTP_EXPOSE=%s", httpPort, httpExpose)
-			return httpPort
-		}
+// GetPrimaryRouterHTTPPort returns app's router primary http port
+// This can be port 80 (default) or the item in HTTP_EXPOSE
+// matching the default router port. If there is no port, returns ""
```

Comment:
If there is no port, then what can happen, how DDEV will act in this case, because it is not obvious to understand?

Reply:
If no port, returns empty string, as you point out. Caller has to do what is required.  For example, in https://github.com/rfay/ddev/blob/d7edc40b90f251604edce0e0e6f93d9d9dacce10/pkg/ddevapp/ddevapp.go#L2998-L3011 the logic is adjusted if there is no value for the returned primary port.